crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -178,6 +201,99 @@ fn create_diagnostic(
     );
 }
 
+fn create_diagnostic_jsx(
+    ctx: &LintContext,
+    function_id: &BindingIdentifier,
+    property: &BindingProperty,
+) {
+    let is_exported = ctx
+        .semantic()
+        .symbols()
+        .get_flags(function_id.symbol_id.get().expect("`symbol_id` should be set"))
```

Comment:
could we replace these `expect` calls with a `let-if-some` block instead? it would be preferable to return early and have potential false positives / false negatives than for `oxlint` to crash on a given rule. this applies anywhere to the other places we are using `expect` too. even if it's safe or guaranteed now, it might not be in the future with refactoring.

something like this:
```rs
let Some(symbol_id) = function_id.symbol_id.get() else {
  return;
}
```

Reply:
@camchenry 
I’ll update the expect and unwrap calls to return instead!
Outside the changes in this PR, there are other places in the codebase for this rule that also use expect. Would it be better to include those in this PR as well?
If that approach is acceptable, I’m considering separating the changes outside this PR into a refactoring PR.

Reply:
aren't symbol ids always going to be set?

maybe debug assert that it's defined

feel free to ignore in this or

Reply:
I've replaced all instances of expect and unwrap with early returns. Commit: c9e1bb5bde3be2fd1aca63a621c26d69b2cd30d0